x/ibchooks/marker_hooks.go

```diff
@@ -0,0 +1,226 @@
+package ibchooks
+
+import (
+	"encoding/json"
+	"strconv"
+	"strings"
+
+	sdkerrors "cosmossdk.io/errors"
+
+	sdktypes "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	transfertypes "github.com/cosmos/ibc-go/v6/modules/apps/transfer/types"
+	"github.com/cosmos/ibc-go/v6/modules/core/exported"
+	ibckeeper "github.com/cosmos/ibc-go/v6/modules/core/keeper"
+	tendermintclient "github.com/cosmos/ibc-go/v6/modules/light-clients/07-tendermint/types"
+
+	"github.com/provenance-io/provenance/x/ibchooks/types"
+	markerkeeper "github.com/provenance-io/provenance/x/marker/keeper"
+	markertypes "github.com/provenance-io/provenance/x/marker/types"
+)
+
+type MarkerHooks struct {
+	MarkerKeeper *markerkeeper.Keeper
+}
+
+func NewMarkerHooks(markerkeeper *markerkeeper.Keeper) MarkerHooks {
+	return MarkerHooks{
+		MarkerKeeper: markerkeeper,
+	}
+}
+
+// ProperlyConfigured returns false when marker hooks are configured incorrectly
+func (h MarkerHooks) ProperlyConfigured() bool {
+	return h.MarkerKeeper != nil
+}
+
+// AddUpdateMarker will add or update ibc Marker with transfer authorities
+func (h MarkerHooks) AddUpdateMarker(ctx sdktypes.Context, packet exported.PacketI, ibcKeeper *ibckeeper.Keeper) error {
+	var data transfertypes.FungibleTokenPacketData
+	if err := json.Unmarshal(packet.GetData(), &data); err != nil {
+		return err
+	}
+	ibcDenom := MustExtractDenomFromPacketOnRecv(packet)
+	if strings.HasPrefix(ibcDenom, "ibc/") {
```

Comment:
I would actually consider this the happy path because it contains the most logic. So you can remove one level of indentation by making this 

```
if !strings.HasPrefix(ibcDenom, "ibc/") {
  return nil
}
```